opentracing-util/src/main/java/io/opentracing/util/GlobalTracer.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.util;
+
+import io.opentracing.NoopTracer;
+import io.opentracing.NoopTracerFactory;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Forwards all methods to another tracer that can be configured by calling {@link #register(Tracer)}.
+ * <p>
+ * When the tracer is needed it delegates all tracing methods to
+ * either the <em>registered</em> global tracer or the {@link NoopTracer}.
+ */
+public final class GlobalTracer implements Tracer {
+    private static final Logger LOGGER = Logger.getLogger(GlobalTracer.class.getName());
+
+    /**
+     * Singleton instance.
+     * <p>
+     * Since we cannot prevent people using {@linkplain #get() GlobalTracer.get()} as a constant,
+     * this guarantees that references obtained before, during or after initialization
+     * all behave as if obtained <em>after</em> initialization once properly initialized.<br>
+     * As a minor additional benefit it makes it harder to circumvent the {@link Tracer} API.
+     */
+    private static final GlobalTracer INSTANCE = new GlobalTracer();
+
+    /**
+     * The resolved {@link Tracer} to delegate to or the {@link NoopTracer} if none was registered yet.
+     * Never {@code null}.
+     */
+    private static volatile Tracer globalTracer = NoopTracerFactory.create();
```

Comment:
To avoid doubt I created a unit test which failed (allowed multiple register() calls to make it through).
The simple solution was to make `register()` synchronized and remove `volatile` as indeed no longer necessary.